cmd/env.go

```diff
@@ -0,0 +1,49 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-node/node"
+)
+
+type Env struct {
+	ndType    node.Type
+	storePath string
+
+	opts []node.Option
+}
+
+func WithEnv(ctx context.Context) context.Context {
+	_, err := GetEnv(ctx)
+	if err == nil {
+		panic("cmd: only one Env is allowed to be set in a ctx")
+	}
+
+	return context.WithValue(ctx, envKey, &Env{})
+}
+
+func GetEnv(ctx context.Context) (*Env, error) {
+	env, ok := ctx.Value(envKey).(*Env)
+	if !ok {
+		return nil, fmt.Errorf("cmd: Env is not set in ctx.Context")
+	}
+
+	return env, nil
+}
+
+func (env *Env) SetNodeType(tp node.Type) {
+	env.ndType = tp
+}
+
+func (env *Env) Options() []node.Option {
+	return env.opts
+}
+
+func (env *Env) addOption(opt node.Option) {
+	env.opts = append(env.opts, opt)
+}
+
+type envCtxKey struct{}
+
+var envKey = envCtxKey{}
```

Comment:
Please add a comment in the code of why this is done. I still think this is unnecessary even with the clearer name now.

Reply:
It is unnecessary in the same level as preallocating slice when you can just append. Preallocating things is a mindset.

Reply:
Added more docs and for this case as well

Reply:
> It is unnecessary in the same level as preallocating slice when you can just append. 

a slice, especially if large, is not comparable an empty struct.

Docs looks good. thank you.

Reply:
Yeah, lets go ban global vars from Go 😁

(its an anti-pattern but there are cases where are they acceptable. An anti-pattern is more for a shared state, in this case envKey has immutable properties)

Reply:
I agree this addition is mostly confusing more than it is helpful. Can you please remove it? This code is used once during node start-up and is not worth our time.

Reply:
...

Reply:
Removed

Reply:
Let's agree that it is at least a code-smell in general. And yes, there are exceptions where it makes sense. But I think we can also agree that this is not one of these exceptions.